model/annotation.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package model
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+)
+
+const (
+	annotationMinLen        = 3
+	annotationMaxLen        = 64
+	annotationAllowedFormat = "annotations must start with letter and can contain only lowercase letters, numbers or '_', '-' characters"
+)
+
+var annotationRegex = regexp.MustCompile("^[a-z].[a-z0-9_-]*$")
```

Comment:
Do we want to support `=` as another valid character so we can have an annotation like `zone=us-west`?

Reply:
Good question. I thought about extending annotations to also contain the value, but to be honest I think it would be better to do it on the database level. We could achieve it quite simply by adding the `Value` column to the `ClusterAnnotations`/`InstallationAnnotations` tables but until then, I think it is better to keep the `=` out of the annotation name so that we do not end up in the situation like having annotation `zone=us-west` with value `us-west` 😄

Reply:
This regex isn't correct; the `.` in the center means "any character," not a literal period. The period needs to be escaped and a test case needs to be added that includes an incorrect separator character. The regex also allows strings that end in a period with no other characters after, which doesn't seem to be intended. It also forces that the annotation start with exactly one lower case letter, a-z, followed by any character (intended to be a period), followed by an optional string of characters (no periods).

Here are some examples of the current regex behaving in an unintended way:
```
gore> regexp.MustCompile("^[a-z].[a-z0-9_-]*$").MatchString("a?b")
true
gore> regexp.MustCompile("^[a-z].[a-z0-9_-]*$").MatchString("aa.b")
false
gore> regexp.MustCompile("^[a-z].[a-z0-9_-]*$").MatchString("a.")
true
gore> regexp.MustCompile("^[a-z].[a-z0-9_-]*$").MatchString("aaaa.")
false
```

I believe the pattern that we actually want is: at least one lowercase letter, followed by a period, followed by a group of alphanumeric characters with at least one letter, and the string doesn't end in a period. Exactly one period is allowed.

I have written the correct regex for this pattern. Also note that in Go, you want to define regular expression pattern using a literal string with backticks, rather than standard double-quoted strings which allow for special escapes like `\n`, or you will be forced to double-escape special characters that you want to use as a literal, e.g. `\\.` instead of `\.` for a literal period.

The argument to `MustCompile` should be changed to:
```
`^[a-zA-Z]+(\.[a-z0-9_-]+)?$`
```

Example test cases:
```
gore> :import regexp
gore> regexp.MustCompile(`^[a-zA-Z]+(\.[a-z0-9_-]+)?$`).MatchString("aaa?bb.cc")
false
gore> regexp.MustCompile(`^[a-zA-Z]+(\.[a-z0-9_-]+)?$`).MatchString("aaabb.cc")
true
gore> regexp.MustCompile(`^[a-zA-Z]+(\.[a-z0-9_-]+)?$`).MatchString("aaabb.cc(")
false
gore> regexp.MustCompile(`^[a-zA-Z]+(\.[a-z0-9_-]+)?$`).MatchString("a.cc(")
false
gore> regexp.MustCompile(`^[a-zA-Z]+(\.[a-z0-9_-]+)?$`).MatchString("a.cc")
true
gore> regexp.MustCompile(`^[a-zA-Z]+(\.[a-z0-9_-]+)?$`).MatchString("banana")
true
```

Reply:
Thanks for spotting that, regex is in fact incorrect but it is only the period that is wrong. The intention was to allow only `[a-z0-9_-]` characters and force annotation to start with a letter. I have misused `.` instead of a `+` so I believe the correct regex should be just that:
 ```
^[a-z]+[a-z0-9_-]*$
```

Reply:
Adding a `Value` column makes sense to me!